stacks-signer/src/signerdb.rs

```diff
@@ -389,6 +403,71 @@ CREATE TABLE IF NOT EXISTS block_rejection_signer_addrs (
     PRIMARY KEY (signer_addr)
 ) STRICT;"#;
 
+// A lighter blocks table to aid in calculating tenure processing times
+// Will only track the most recent tenure blocks
+static CREATE_TENURE_BLOCKS_TABLE: &str = r#"
+CREATE TABLE IF NOT EXISTS tenure_blocks (
+    signer_signature_hash TEXT NOT NULL PRIMARY KEY,
+    reward_cycle INTEGER NOT NULL,
+    consensus_hash TEXT NOT NULL,
+    proposed_time INTEGER NOT NULL,
+    validation_time_ms INTEGER NOT NULL,
+    stacks_height INTEGER NOT NULL,
+    tenure_change INTEGER NOT NULL
+) STRICT;"#;
+
+// Migration logic necessary to move from blocks into tenure_blocks table
+// It will only migrate globally accepted blocks that are less than 2 reward cycles old.
+static MIGRATE_GLOBALLY_ACCEPTED_BLOCKS_TO_TENURE_BLOCKS: &str = r#"
+INSERT INTO tenure_blocks (
+    signer_signature_hash,
+    reward_cycle,
+    consensus_hash,
+    proposed_time,
+    validation_time_ms,
+    stacks_height,
+    tenure_change
+)
+SELECT 
+    signer_signature_hash,
+    reward_cycle,
+    consensus_hash,
+    json_extract(block_info, '$.proposed_time') AS proposed_time,
+    COALESCE(json_extract(block_info, '$.validation_time_ms'), 0) AS validation_time_ms,
+    stacks_height,
+    json_extract(block_info, '$.tenure_change') AS tenure_change
```

Comment:
I think this line will actually fail in a read DB migration, because `$.tenure_change` doesn't exist. So, you need to do a `COALESCE` at the least

Reply:
This is making me wonder more about how to handle this migration - if we migrate mid-tenure, can we just default to `true`, and so then the last pre-migration block would be the start of the idle timer?

Reply:
KK will drop the table in favour of a new one that extracts out the relevant info from block info. Will make the migration easier at the very least and then will just retreieve the blocks up to the last tenure change true.

Reply:
Done.